splice_machine/src/test/java/com/splicemachine/derby/utils/ShowCreateTableIT.java

```diff
@@ -382,4 +397,93 @@ public void testTextDefaultUserRole() throws Exception
+                ") \n" +
+                "ROW FORMAT DELIMITED FIELDS TERMINATED BY ','\n" +
+                "STORED AS TEXTFILE\n" +
+                "LOCATION '"+getExternalResourceDirectory()+"testCsvFile';", rs.getString(1));
+    }
+
+    @Category(HBaseTest.class)
+    @Test
+    public void testExternalTableParquetWithoutCompression() throws Exception {
+        //Parquet Without compression
+        String parquetDDL = String.format("create external table SHOWCREATETABLEIT.testParquet (col1 int, col2 varchar(24))" +
+                "partitioned by (col1) STORED AS parquet LOCATION '%s'", getExternalResourceDirectory() + "testParquet");
+
+        methodWatcher.executeUpdate(parquetDDL);
+        ResultSet rs = methodWatcher.executeQuery("call syscs_util.SHOW_CREATE_TABLE('SHOWCREATETABLEIT','TESTPARQUET')");
+        rs.next();
+        Assert.assertEquals("CREATE EXTERNAL TABLE \"SHOWCREATETABLEIT\".\"TESTPARQUET\" (\n" +
+                "\"COL1\" INTEGER\n" +
+                ",\"COL2\" VARCHAR(24)\n" +
+                ") \n" +
+                "PARTITIONED BY (COL1)\n" +
+                "STORED AS PARQUET\n" +
+                "LOCATION '"+getExternalResourceDirectory()+"testParquet';", rs.getString(1));
+    }
+
+    @Category(HBaseTest.class)
+    @Test
+    public void testExternalTableOrcSnappy() throws Exception {
+        //Orc With compression
+        String orcDDL = String.format("create external table SHOWCREATETABLEIT.testOrcSnappy (col1 int, col2 varchar(24))" +
+                "compressed with snappy partitioned by (col2) STORED AS ORC LOCATION '%s'", getExternalResourceDirectory()+"testOrcSnappy");
+        methodWatcher.executeUpdate(orcDDL);
+        ResultSet rs = methodWatcher.executeQuery("call syscs_util.SHOW_CREATE_TABLE('SHOWCREATETABLEIT','TESTORCSNAPPY')");
+        rs.next();
+        Assert.assertEquals("CREATE EXTERNAL TABLE \"SHOWCREATETABLEIT\".\"TESTORCSNAPPY\" (\n" +
+                "\"COL1\" INTEGER\n" +
+                ",\"COL2\" VARCHAR(24)\n" +
+                ") \n" +
+                "COMPRESSED WITH snappy\n" +
+                "PARTITIONED BY (COL2)\n" +
+                "STORED AS ORC\n" +
+                "LOCATION '"+getExternalResourceDirectory()+"testOrcSnappy';", rs.getString(1));
+    }
+
+    @Test
+    public void testSystemTable() throws Exception {
+        try
+        {
+            methodWatcher.execute("call syscs_util.SHOW_CREATE_TABLE('SYS','SYSTABLES')");
+            Assert.fail("Expected to fail");
+        }
+        catch (SQLException e)
+        {
+            String sqlState = e.getSQLState();
+            Assert.assertEquals("42X62",sqlState);
+        }
+    }
+
+    @Test
+    public void testView() throws Exception {
+        try
+        {
+            methodWatcher.executeUpdate("create view v1 as (select a1 from showcreatetableit.t1)");
+            methodWatcher.execute("call syscs_util.SHOW_CREATE_TABLE('SYS','SYSTABLES')");
```

Comment:
Did you mean to call this on the view instead of the system table?

Reply:
Thanks for catching it!